shared/src/commonMain/kotlin/com/mbta/tid/mbta_app/model/AlertSummary.kt

```diff
@@ -113,38 +115,268 @@ data class AlertSummary(
+                                        earlier + branched
+                                    )
+                                }
+                        } else emptyList()
+                    )
+                    .mapNotNull { (pattern, stopIds) ->
+                        val stopIdsOnPattern =
+                            stopIds
+                                ?.filter { stopOnTrip ->
+                                    alert.anyInformedEntitySatisfies {
+                                        // `affectedStops` only includes parent stops, here we check
+                                        // if the child stops on each pattern are affected
+                                        checkStop(stopOnTrip)
+                                        checkRoute(pattern.routeId)
+                                    }
+                                }
+                                ?.mapNotNull { global.stops[it]?.resolveParent(global)?.id }
+
+                        if (stopIdsOnPattern != null) pattern to stopIdsOnPattern else null
+                    }
+                    .filter { (_, affectedStopsOnPattern) -> affectedStopsOnPattern.isNotEmpty() }
+                    .toMap()
 
-                    if (stopIdsOnPattern != null) pattern to stopIdsOnPattern else null
+            // On the D branch, there are patterns that terminate at Reservoir and Riverside, this
+            // will remove stop lists that are subsets of some other stop list so that we don't
+            // display "Westbound stops" instead of "Riverside" in this case.
+            return if (patternStops.size > 1)
+                patternStops.filter {
+                    !patternStops.any { (otherPattern, otherStops) ->
+                        otherPattern != it.key &&
+                            otherStops.size > it.value.size &&
+                            otherStops.containsAll(it.value)
+                    }
                 }
-                .filter { (_, affectedStopsOnPattern) -> affectedStopsOnPattern.isNotEmpty() }
-                .toMap()
+            else patternStops
         }
+
+        // The first value in these pairs is the list of trunk stops for each route, including a few
+        // minor child stop differences at some stops, like Park and Kenmore. These trunk lists
+        // start at the first trunk stop on the opposite end of the trip (Lechmere and Kenmore).
+        // The second value contains all the child stops that exist only on each branch.
+        // These are hard coded because the patterns provided to `summarizing` will only include
+        // ones served at the selected stop, they don't take other branches into account, but we
+        // always want to show when a disruption is happening on all downstream branches.
+        private val westboundBranches =
+            listOf(
+                Pair( // B Branch
+                    listOf(
+                        "70502",
```

Comment:
suggestion (non-blocking): Maybe worth commenting the first / last stop of each segment just to help with quickly understanding each list